### PR TITLE
Hyperlink URLs

### PR DESCRIPTION
**Pull request by pauljm**

### CODE REVIEW

File: app/models/micropost.rb [hunk 1]
```diff
@@ -4,6 +4,11 @@ class Micropost < ActiveRecord::Base
   validates :content, presence: true, length: { maximum: 140 }
   validates :user_id, presence: true
 
+  def html_content
+    html = CGI::escapeHTML( content )
+    html = html.gsub(/(https?:\/\/[-A-Za-z0-9\.]+)/, '<a href="\1">\1</a>')
```
Comment (pauljm) [inline app/models/micropost.rb]: Have you considered simple string sub?

**Comment by railstutorial** [comment]
I don't want to merge this with the main sample app, but you're moving in the direction of some nice enhancements. One suggestion is to familiarize yourself with pre-existing helper methods so you don't have to reinvent the wheel. For example, you can use [`html_escape`](http://apidock.com/rails/ERB/Util/html_escape/class) (or `h` for short) to escape content, [`sanitize`](http://apidock.com/rails/ActionView/Helpers/SanitizeHelper/sanitize) to strip out dangerous tags, and the [autolink](https://rubygems.org/gems/rails_autolink) gem to link things like http://example.com/ automatically.

**Comment by pauljm** [comment]
Thanks a lot for the notes and sorry for the trouble. I opened this PR as
part of a demo and forgot to close it right after.

Thanks for a great sample project.

--
831-238-4370

*Want to tap into my network? Join me on Conspire
<https://www.conspire.com/welcome?utm_source=emailsignature&utm_medium=email&utm_content=paulmcreynolds&utm_campaign=email%20signatures>*

**Comment by mhartl** [comment]
No worries! It's great to see people learning things. :-)

**Pull request closed without merging**